src_c/event.c

```diff
@@ -99,6 +99,12 @@ static SDL_TimerID _pg_repeat_timer = 0;
 static SDL_Event _pg_repeat_event;
 static SDL_Event _pg_last_keydown_event = {0};
 
+#define NUM_KEYS 512
+
+int held_down_keys[NUM_KEYS];
+int released_keys[NUM_KEYS];
+int keys_being_held_down[NUM_KEYS];
```

Comment:
This is nitpicky, but since you're only using 1 bit of information, this could use less memory by using a smaller integer type than int. Ints are 4 bytes (or 8 sometimes?), chars are only 1 byte, so you could do this as an array of chars. Maybe with a comment explaining that it isn't text, but an array of "bools."